gui/src/app/tinystan/index.ts

```diff
@@ -280,17 +305,35 @@ export default class StanModel {
       );
 
       // copy out parameters of interest
-      const draws: number[][] = Array.from({ length: n_params }, () => []);
-      for (let i = 0; i < n_draws; i++) {
-        for (let j = 0; j < n_params; j++) {
-          const elm = out_buffer[i * n_params + j];
-          draws[j][i] = elm;
-        }
-      }
+      const draws: number[][] = Array.from({ length: n_params }, (_, i) =>
+        Array.from({ length: n_draws }, (_, j) => out_buffer[i + n_params * j])
+      );
+
       // Clean up
       this.m._free(out_ptr);
 
-      return { paramNames, draws };
+      let metric_array: number[] | number[][] | null = null;
+
+      if (save_metric) {
+        if (metric === HMCMetric.DENSE) {
+          const metric_buffer = this.m.HEAPF64.subarray(
+            metric_out / Float64Array.BYTES_PER_ELEMENT,
+            metric_out / Float64Array.BYTES_PER_ELEMENT + free_params * free_params,
+          );
+          metric_array = Array.from({ length: free_params }, (_, i) =>
+            Array.from({ length: free_params }, (_, j) => metric_buffer[i * free_params + j])
```

Comment:
And here it's "times i plus j"

Reply:
If it's performance-impactful to the level that row-major/column-major choice will be noticeable, you may actually want something from either the [math.js](https://mathjs.org/) library or an [ArrayBuffer](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/ArrayBuffer) object.

But isn't this pretty much just getting passed to the sampler worker, which will copy it (once) into memory managed by the WASM code?

Reply:
This is output, so it really depends on what we end up doing with it later. The input equivalent to this is the `init_inv_metric` argument that I've left unimplemented for now, and it's certainly true that something like math.js would be helpful there

Reply:
Cool. Sounds like something we can worry about once it's used for something other than just display.